mesonbuild/dependencies/ui.py

```diff
@@ -252,7 +252,7 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
 
 sdl2_factory = DependencyFactory(
     'sdl2',
-    [DependencyMethods.PKGCONFIG, DependencyMethods.CONFIG_TOOL, DependencyMethods.EXTRAFRAMEWORK],
+    [DependencyMethods.PKGCONFIG, DependencyMethods.CONFIG_TOOL, DependencyMethods.EXTRAFRAMEWORK, DependencyMethods.CMAKE],
     configtool_class=SDL2DependencyConfigTool,
```

Comment:
```suggestion
    configtool_class=SDL2DependencyConfigTool,
    cmake_name='SDL2',
```

Reply:
This currently does not work on my machine. As you noted -- which I was unaware of -- given the dependency name "SDL2" cmake will also look up the lowercase version in config mode, so that does mean that the sdl2 project has a canonical name for find_package() that works everywhere.

But, the default name in our dependency factory is lowercase, not uppercase. And cmake doesn't look up the uppercasePackageName. :)

This "works" on Windows, because on Windows filesystems attempting to open and read the file `sdl2Config.cmake` will actually successfully open and read the file that actually exists, which is `SDL2Config.cmake`. Yay for case insensitive filesystems...

Fortunately, the difference between exact dependency names across dependency methods is a common issue and thus has builtin overrides, you can use `<method>_name='override-name'` if it needs to use a different name specific to cmake.

Reply:
Thanks for pointing this out, it's really useful (as I'm not familiar with the codebase).